pkg/validate/image.go

```diff
@@ -0,0 +1,106 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package validate
+
+import (
+	"github.com/kubernetes-incubator/cri-tools/pkg/framework"
+	internalapi "k8s.io/kubernetes/pkg/kubelet/api"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+var (
+	// image name for test image api
+	testImageName = "busybox"
+
+	// name-tagged reference for latest test image
+	latestTestImageRef = testImageName + ":latest"
+
+	// Digested reference for test image
+	busyboxDigestRef = testImageName + "@sha256:a59906e33509d14c036c8678d687bd4eec81ed7c4b8ce907b888c607f6a1e0e6"
+)
+
+var _ = framework.KubeDescribe("Image Manager", func() {
+	f := framework.NewDefaultCRIFramework()
+
+	var c internalapi.ImageManagerService
+
+	BeforeEach(func() {
+		c = f.CRIClient.CRIImageClient
+	})
+
+	It("public image with tag should be pulled and removed [Conformance]", func() {
+		framework.TestPullPublicImage(c, latestTestImageRef)
+	})
+
+	It("public image without tag should be pulled and removed [Conformance]", func() {
+		framework.TestPullPublicImage(c, testImageName)
+	})
+
+	It("public image with digest should be pulled and removed [Conformance]", func() {
+		framework.TestPullPublicImage(c, busyboxDigestRef)
+	})
+
+	It("image status get image fields should not be empty [Conformance]", func() {
+		framework.PullPublicImage(c, latestTestImageRef)
+
+		defer framework.RemoveImage(c, latestTestImageRef)
+
+		status := framework.ImageStatus(c, latestTestImageRef)
+		Expect(status.Id).NotTo(BeNil(), "image Id should not be nil")
+		Expect(len(status.RepoTags)).NotTo(Equal(0), "should have repoTags in image")
+		Expect(status.Size_).NotTo(BeNil(), "image Size should not be nil")
+	})
+
+	It("ListImage should get exactly 3 image in the result list [Conformance]", func() {
```

Comment:
It may fail if runtime isn't a fresh install. I think it's better to check those images are included in the image list.